hogben/simulate.py

```diff
@@ -1,101 +1,156 @@
+        path = importlib_resources.files('hogben.data.directbeams'
+                                         ).joinpath(pol_instr[inst_or_path])
 
-def simulate_magnetic(sample, angle_times, scale=1, bkg=5e-7, dq=2,
-                      mm=True, mp=True, pm=True, pp=True, angle_scale=0.7,
-                      directbeam_path=None):
+    return path
+
+
+def simulate_magnetic(sample: Union['refnx.reflect.Stucture',
+                                    'refl1d.model.Stack'],
+                      angle_times: np.ndarray, scale: float = 1.0,
+                      bkg: float = 5e-7, dq: float = 2, mm: bool = True,
+                      mp: bool = True, pm: bool = True, pp: bool = True,
+                      angle_scale: float = 0.3,
+                      inst_or_path: str = 'OFFSPEC') -> tuple:
     """Simulates an experiment of a given magnetic `sample` measured
        over a number of angles.
 
     Args:
-        sample (refnx.reflect.Stucture or
-                refld1d.model.Stack): sample to simulate.
-        angle_times (list): points and times for each angle to simulate.
-        scale (float): experimental scale factor.
-        bkg (float): level of instrument background noise.
-        dq (float): instrument resolution.
-        pp (bool): whether to simulate "plus plus" spin state.
-        pm (bool): whether to simulate "plus minus" spin state.
-        mp (bool): whether to simulate "minus plus" spin state.
-        mm (bool): whether to simulate "minus minus" spin state.
-        angle_scale (float): angle to use when scaling directbeam flux.
-        directbeam_path (str): path to directbeam file to simulate with.
+        sample: sample structure to simulate.
+        angle_times: array of angle, number of data points and time to measure
+        for each angle to simulate. e.g. [(0.7, 100, 5), (2.0, 100, 20)]
+        scale: experimental scale factor.
+        scale: experimental scale factor.
+        bkg: level of instrument background noise.
+        dq: instrument resolution.
+        pp: whether to simulate "plus plus" spin state.
+        pm: whether to simulate "plus minus" spin state.
+        mp: whether to simulate "minus plus" spin state.
+        mm: whether to simulate "minus minus" spin state.
+        angle_scale: angle to use when scaling directbeam flux.
+        inst_or_path: dictionary entry of direct beam stored in hogben or
+        filepath of directbeam file to simulate with..
 
     Returns:
         tuple: model and simulated data for the given `sample`.
 
     """
     models, datasets = [], []
 
-    # Default path to the directbeam file for OFFSPEC when polarised.
-    if directbeam_path is None:
-        directbeam_path = importlib_resources.files('hogben.data.directbeams').joinpath('OFFSPEC_polarised_old.dat')
-        angle_scale = 0.3
+    direct_beam = direct_beam_path(inst_or_path, polarised=False)
+
+    if not direct_beam:  # Local filepath provided so direct_beam_path==None
```

Comment:
Could you have this bit of logic in the `direct_beam_path` function instead? So that if it's passed a path it just returns it rather than None? That would make behaviour more consistent as it would always return a path

There may be a reason this doesn't make sense though!